File: app/src/main/java/com/tistory/ykyahwa/kotlingithubbooksample/api/GithubApi.java
package com.tistory.ykyahwa.kotlingithubbooksample.api;

import com.tistory.ykyahwa.kotlingithubbooksample.api.model.GithubRepo;
import com.tistory.ykyahwa.kotlingithubbooksample.api.model.RepoSearchResponse;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface GithubApi {

    @GET("search/repositories")
    Call<RepoSearchResponse> searchRepository(@Query("q") String query);

    @GET("repos/{owner}/{name}")
    Call<GithubRepo> getRepository(@Path("owner") String ownerLogin, @Path("name") String repoName);
}

File: app/src/main/kotlin/com/tistory/ykyahwa/kotlingithubbooksample/api/GithubApi.kt
package com.tistory.ykyahwa.kotlingithubbooksample.api

import com.tistory.ykyahwa.kotlingithubbooksample.api.model.GithubRepo
import com.tistory.ykyahwa.kotlingithubbooksample.api.model.RepoSearchResponse
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface GithubApi {

    @GET("search/repositories")
    fun searchRepository(@Query("q") query: String): Observable<RepoSearchResponse>

    @GET("repos/{owner}/{name}")
    fun getRepository(@Path("owner") ownerLogin: String, @Path("name") repoName: String): Observable<GithubRepo>
}
